util/network-devp2p/src/discovery.rs

```diff
@@ -80,15 +80,28 @@ impl BucketEntry {
 	}
 }
 
-pub struct NodeBucket {
-	nodes: VecDeque<BucketEntry>, //sorted by last active
+struct FindNodeRequest {
+	// Time when the request was sent
+	sent_at: Instant,
+	// Number of items sent by the node
+	response_count: usize,
+	// Whether the request have been answered yet
+	answered: bool,
 }
 
-struct PendingRequest {
-	packet_id: u8,
+struct PingRequest {
+	// Time when the request was sent
 	sent_at: Instant,
-	packet_hash: H256,
-	response_count: usize, // Some requests (eg. FIND_NODE) have multi-packet responses
+	// The node to which the request was sent
+	node: NodeEntry,
+	// The hash sent in the Ping request
+	echo_hash: H256,
+	// The hash Parity used to respond with
```

Comment:
Can you include in the comment the Parity version that this was fixed in so this can be removed eventually?

Reply:
I think it was added in https://github.com/paritytech/parity-ethereum/pull/8042 and shipped in 1.11.0. AFAIK geth just ignores this value.

Reply:
I'm pretty see Geth checks the hash. [See here](https://github.com/ethereum/go-ethereum/blob/c73b654fd172464f598c239b4e7c4bc07a641e3b/p2p/discover/udp.go#L292).